packages/docs/site/docs/11-architecture/18-host-your-own-playground.md

```diff
@@ -0,0 +1,169 @@
+# Host your own Playground
+
+You can host the Playground on your own domain instead of `playground.wordpress.net`.
+
+This is useful for having full control over its content and behavior, as well as removing dependency on a third-party remote endpoint. It can provide a customized user experience, for example: preinstalled plugins, default theme, site settings, and demo content.
+
+Also, if you expect to see heavy traffic, it's good etiquette to provide your own server/CDN resources to serve the static file assets, rather than relying on the `wordpress.net` server.
+
+#### Usage
+
+A self-hosted Playground can be embedded as an iframe.
+
+```html
+<iframe src="https://my-playground.com"></iframe>
+```
+
+Or dynamically loaded by passing the remote URL to the Playground Client.
+
+```ts
+import { startPlaygroundWeb } from '@wp-playground/client';
+
+const client = await startPlaygroundWeb({
+	iframe: document.getElementById('wp'),
+	remoteUrl: `https://my-playground.com/remote.html`,
+});
+```
+
+## Build static assets
+
+Create a shallow clone of the Playground repository.
+
+```sh
+git clone -b trunk --single-branch --depth 1 git@github.com:WordPress/wordpress-playground.git
+```
+
+Install dependencies, and build the website.
+
+```sh
+npm install
+npm run build:website
+```
+
+This command internally runs the `nx` task `build:wasm-wordpress-net`. It copies the built assets from packages `remote` and `website` into a new folder at the path.
+
+```
+dist/packages/playground/wasm-wordpress-net
+```
+
+The entire service of the Playground consists of the content of this folder.
+
+It includes:
+
+-   Data and WASM files for all available PHP and WordPress versions
+-   `remote.html` - the core of Playground
+-   `index.html` - the shell, or browser chrome
+-   Web Worker script
+
+You can deploy the content of the folder to your server using SSH, such as `scp` or `rsync`.
+
+It is a static site, except for these dynamic aspects.
+
+-   Apache server directive `.htaccess` file - the result of combining `.htaccess` from the packages `remote` and `website`
+-   Plugin download proxy, `plugin-proxy.php`
+
+For these to work, you need a server environment with Apache and PHP installed.
+
+## NGINX configuration
+
+As an alternative to Apache, here is an example of using NGINX to serve the Playground.
+
+The Apache `.htaccess` file looks like this.
+
+```htaccess
+AddType application/wasm .wasm
+AddType	application/octet-stream .data
+
+<FilesMatch "iframe-worker.html$">
+  Header set Origin-Agent-Cluster: ?1
+</FilesMatch>
+
+Header set Cross-Origin-Resource-Policy: cross-origin
+Header set Cross-Origin-Embedder-Policy: credentialless
```

Comment:
They are needed for now, but I'd like to get rid of them eventually: https://github.com/WordPress/wordpress-playground/issues/411

Reply:
These lines come from `packages/playground/remote/public/.htaccess`.

@dmsnell I agree with your concern that this and other parts of the documentation will easily get out of date, if we have to remember to update it at the same time as the code.

Maybe a solution is to use some kind of template function or variable to pull this content from the source code directly, so when the source changes, it is automatically reflected in the docs. That still won't solve the derived content, the NGINX equivalent of this `.htaccess`, which someone has to remember to update manually.

Reply:
Since we're using docusaurus and mdx, that should be possible. But I'd much rather just get rid of most .htaccess rules

Reply:
> use some kind of template function or variable to pull this content from the source code directly, so when the source changes, it is automatically reflected in the docs.

we'd probably end up with three things to maintain then 🙃

it's okay. I feel like this is probably fine. they are docs, after all, so at least they won't break things.

Reply:
For now, I think the simplest low-tech solution is to add a note in the docs about the possibility that this `.htaccess` and NGINX config **example could be out of date**; and include a link to the source file, i.e., the .htaccess file in `remote` and `website` packages. That way the reader can refer to them as the source of truth.

Keeping the docs in sync with actual information seems like a common question of workflow. This week I saw an example of this, in the README file of the VS Code extension.

> [@adamziel](https://github.com/adamziel) and [@danielbachhuber](https://github.com/danielbachhuber/) have permissions to publish.

https://github.com/WordPress/playground-tools/tree/trunk/packages/vscode-extension#publishing

For this sentence to be automatically updated, the names could be replaced with a dynamically generated list. But I also wonder if it's worth the time/effort to automate it. ([xkcd#1205](https://xkcd.com/1205/))

Reply:
Resolved in commit [cde9166](https://github.com/WordPress/wordpress-playground/pull/574/commits/cde9166e3aebac615a5c3fbf72a1184328e9d83d).

It adds a note in the section NGINX Configuration that the example may go out of date, and a recommendation to refer to the source files with links to the Apache `.htaccess` files.